As we are using GPLv2 only networkmanager must be patched due to readline5 being used, which does not provide rl_echo_signal_char
diff -aurN a/src/nmcli/common.c b/src/nmcli/common.c
--- a/src/nmcli/common.c	2022-02-23 16:40:42.000000000 +0100
+++ b/src/nmcli/common.c	2024-01-12 12:07:42.656684945 +0100
@@ -905,15 +905,6 @@
     while (!rl_got_line && g_main_loop_is_running(loop) && !nmc_seen_sigint())
         g_main_context_iteration(NULL, TRUE);
 
-    /* If Ctrl-C was detected, complete the line */
-    if (nmc_seen_sigint()) {
-        rl_echo_signal_char(SIGINT);
-        if (!rl_got_line) {
-            rl_stuff_char('\n');
-            rl_callback_read_char();
-        }
-    }
-
     /* Add string to the history */
     if (rl_string && *rl_string)
         add_history(rl_string);
